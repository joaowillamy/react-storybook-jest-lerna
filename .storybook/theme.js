import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
