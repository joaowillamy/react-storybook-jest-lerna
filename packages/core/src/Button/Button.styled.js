import styled, { css } from "styled-components";

export const Button = styled.button`
  &,
  &:link,
  &:visited {
    text-decoration: none;
    padding: 0 20px;
    display: inline-block;
    transition: all 0.2s;
    margin: 5px;
    cursor: pointer;
    align-items: center;
    justify-content: center;
    height: 42px;
    font-size: 14px;
    border-radius: 5px;
    font-weight: 600;
  }

  &:hover {
    ${({ disabled, theme }) =>
      !disabled &&
      css`
        box-shadow: ${theme.shadows.md()};
      `}
  }

  &:active,
  &:focus {
    outline: none;
    box-shadow: ${({ theme }) => theme.shadows.sm()};
  }

  &:disabled,
  & [disabled] {
    background-color: #c9cdce;
    color: #80898b;
    cursor: not-allowed;
    border: 1px solid #a5abac;
  }

  ${({ variant, theme, disabled }) => {
    if (variant === "primary" && !disabled) {
      return css`
        border: 1px solid ${theme.colors.yellowSecondary};
        background-color: ${theme.colors.yellowPrimary};
        color: ${theme.colors.black};
      `;
    } else if (variant === "secondary" && !disabled) {
      return css`
        border: 1px solid ${theme.colors.bluePrimary};
        background-color: #fff;
        color: ${theme.colors.blueSecondary};
      `;
    }
  }}
`;
