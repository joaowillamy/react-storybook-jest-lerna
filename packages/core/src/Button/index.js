import React from "react";
import PropTypes from "prop-types";

import * as S from "./Button.styled";

const Button = props => {
  return <S.Button {...props} />;
};

Button.propTypes = {
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(["primary", "secondary"])
};

Button.defaultProps = {
  disabled: false,
  variant: "primary"
};

export default Button;
