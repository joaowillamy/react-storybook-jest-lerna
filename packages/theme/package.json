{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
