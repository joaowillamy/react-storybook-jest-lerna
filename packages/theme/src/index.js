export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
