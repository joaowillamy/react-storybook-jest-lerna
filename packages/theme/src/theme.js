const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
