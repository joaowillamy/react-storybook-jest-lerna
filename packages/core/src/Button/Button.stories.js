import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Button from ".";

export default {
  component: Button,
  title: "Core|Button"
};

export const primary = () => (
  <ThemeProvider>
    <Button>{"Button"}</Button>
  </ThemeProvider>
);

export const secondary = () => (
  <ThemeProvider>
    <Button variant="secondary">{"Button"}</Button>
  </ThemeProvider>
);

export const disabled = () => (
  <ThemeProvider>
    <Button disabled>{"Button"}</Button>
  </ThemeProvider>
);
