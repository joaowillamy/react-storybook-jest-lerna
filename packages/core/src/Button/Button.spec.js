import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Button from ".";

describe("Button", () => {
  test("primary", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Button>{"Test"}</Button>
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("secondary", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Button variant="secondary">{"Test"}</Button>
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
