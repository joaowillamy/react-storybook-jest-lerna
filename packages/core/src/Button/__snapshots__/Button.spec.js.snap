// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button primary 1`] = `
.c0 {
  border: 1px solid #DE9E1F;
  background-color: #FDCB13;
  color: #1F2D30;
}

.c0,
.c0:link,
.c0:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 20px;
  display: inline-block;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  margin: 5px;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 42px;
  font-size: 14px;
  border-radius: 5px;
  font-weight: 600;
}

.c0:hover {
  box-shadow: 0 3px 4px #00000029;
}

.c0:active,
.c0:focus {
  outline: none;
  box-shadow: 0 1px 3px #00000029;
}

.c0:disabled,
.c0 [disabled] {
  background-color: #c9cdce;
  color: #80898b;
  cursor: not-allowed;
  border: 1px solid #a5abac;
}

<button
  className="c0"
  disabled={false}
>
  Test
</button>
`;

exports[`Button secondary 1`] = `
.c0 {
  border: 1px solid #18ACC4;
  background-color: #fff;
  color: #007A8D;
}

.c0,
.c0:link,
.c0:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 20px;
  display: inline-block;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  margin: 5px;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 42px;
  font-size: 14px;
  border-radius: 5px;
  font-weight: 600;
}

.c0:hover {
  box-shadow: 0 3px 4px #00000029;
}

.c0:active,
.c0:focus {
  outline: none;
  box-shadow: 0 1px 3px #00000029;
}

.c0:disabled,
.c0 [disabled] {
  background-color: #c9cdce;
  color: #80898b;
  cursor: not-allowed;
  border: 1px solid #a5abac;
}

<button
  className="c0"
  disabled={false}
>
  Test
</button>
`;
